tests/test_readme.py

```diff
@@ -0,0 +1,41 @@
+import os
+import pathlib
+import re
+import string
+import numpy as np
+import pytest
+import pandas as pd
+from sklearn.datasets import make_classification
+
+PYTHON_SNIPPET_REGEX = re.compile(r"```python([^`]+)```")
+
+
+@pytest.fixture
+def data():
+    n_feats = 10
+    n_samples = 10
+    feature_names = [string.ascii_lowercase[i] for i in range(n_feats)] + ["target"]
+    X, y = make_classification(n_feats, n_samples, random_state=0)
+    data = pd.DataFrame(
+        data=np.hstack((X, y.reshape((n_samples, 1)))), columns=feature_names
+    )
+    return data
+
+
+@pytest.fixture
+def readme_string():
+    readme_dir = pathlib.Path(__file__).parent.parent.absolute()
+    with open(os.path.join(readme_dir, "README.md")) as f:
+        readme = f.read()
+    return readme
+
+
+@pytest.mark.slow
+def test_readme_code_examples(data, readme_string):
+    """ this tests checks that every (almost) python snippet in the README
+    would run sequentially without raising errors"""
+    python_snippets = PYTHON_SNIPPET_REGEX.findall(readme_string)
+    for snippet in python_snippets:
+        if "read_csv" in snippet:  # the file is an unexisting example
+            continue
+        exec(snippet.strip())
```

Comment:
hmmm I found the way you tested this cool, I am only worried about the slow mark. How slow is it? if it is too slow I would prefer the regular way

Reply:
I only know about this. But I guess this is ok too for me
https://docs.python.org/3/library/doctest.html

Reply:
I didn't know doctest, seems very cool. I tried to apply it to our case but it doesn't recognize the snippets unless we put a `>>>` everywhere (not a big problem, maybe even clearer). The biggest problem I see is that we would have to get rid of 
```
>>> import pandas as pd
>>> data = pd.read_csv('test.csv')
```
as it's doomed to fail

Reply:
regarding the slowness, it's 100 seconds, but I think it's well invested. we can be sure that whoever runs those snippets will not encounter an error. it's also less complicated to maintain.

Reply:
hmm ok. we should add some condition to run these on the pipeline then.. Like target branch is main or something